test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+public class Runner {
+    private static final Logger log = Logger.getLogger(Runner.class.getName());
+
+    private static final Os currentOs = detectOs();
+    private static String libExt;
+    private static final Arch currentArch = detectArch();
+    private static final int currentJvmVersion = detectJvmVersion();
+    private static final Jvm currentJvm = detectJvm();
+
+    private static Os detectOs() {
+        String osName = System.getProperty("os.name").toLowerCase();
+        if (osName.contains("linux")) {
+            libExt = "so";
+            return Os.LINUX;
+        } else if (osName.contains("mac")) {
+            libExt = "dylib";
+            return Os.MACOS;
+        } else if (osName.contains("windows")) {
+            libExt = "dll";
+            return Os.WINDOWS;
+        }
+        throw new IllegalStateException("Unknown OS type");
+    }
+
+    private static int detectJvmVersion() { // Not sure if this works for zing
+        String prop = System.getProperty("java.vm.version");
+
+        if (prop.startsWith("25.") && prop.charAt(3) > '0') {
+            return 8;
+        } else if (prop.startsWith("24.") && prop.charAt(3) > '0') {
+            return 7;
+        } else if (prop.startsWith("20.") && prop.charAt(3) > '0') {
+            return 6;
+        } else {
+            String substring = prop.substring(0, prop.indexOf('.'));
+            int possibleVersion = Integer.parseInt(substring);
+            if (possibleVersion < 9) {
+                return 9;
```

Comment:
When is this branch needed? Is it ever reachable?

Reply:
I think this was added for future proofing